charts/trino/templates/tests/test-resource-groups-db.yaml

```diff
@@ -0,0 +1,33 @@
+{{- if eq .Values.resourceGroups.type "properties" }}
+apiVersion: v1
+kind: Pod
+metadata:
+  name: {{ include "trino.fullname" . }}-test-add-resourcegroups-data
+  labels:
+    {{- include "trino.labels" . | nindent 4 }}
+    app.kubernetes.io/component: test
+    test: add-resource-groups-data
+  annotations:
+    "helm.sh/hook": test
+    "helm.sh/hook-weight": "-1"
+spec:
+  containers:
+    - name: mysql-client
+      image: bitnami/mysql:8.1
+      command:
+        - /bin/sh
+        - -c
+        - |
+          echo "Inserting data into MySQL";
```

Comment:
This is not a test - it just bootstraps the service. Maybe do this in `test.sh`? 
You could create a configmap with an `.sql` file, pass it to the mysql chart and it'll be executed when initializing the db. See https://artifacthub.io/packages/helm/bitnami/mysql#customize-a-new-mysql-instance

Reply:
@nineinchnick The database tables required for the resource group manager are only created once Trino is fully initialized. Since MySQL starts before Trino, attempting to insert data in the MySQL initialization phase would fail, as the necessary tables don’t yet exist.

For this reason, this setup inserts data only after Trino has completed its initialization and the table structure is confirmed.

Let me know if you have other ideas or if there’s another approach that might work given this dependency on Trino’s initialization timing.

Reply:
If it's using flyway to apply migrations, initializing the DB outside of Trino will probably be annoying. It's fine to run the setup here then, but you should also do some assertions here too, and not rely on the order of execution of all tests. It's OK to duplicate some stuff in tests.

Reply:
@nineinchnick The database tables are only required when using the resource group manager with a database, specifically for the `resource_groups_properties_values` test case. For all other test cases, MySQL does not require database creation, so I don’t anticipate errors in other test cases due to missing tables.

Therefore, I added `"helm.sh/hook-weight": "99"` to `charts/trino/templates/tests/test-connection.yaml` to ensure that in the `resource_groups_properties_values` test case, Trino completes inserting the required table data before the test runs.

To isolate this dependency and avoid affecting other test cases, I’m considering making `"helm.sh/hook-weight": "99"` conditional so that it only applies in the `resource_groups_properties_values` test case. Would this approach address your concerns?

Reply:
We should not have tests depend on each other. If one test needs additional bootstrapping, we can either put a conditional block in it, or copy the whole test and make both copies conditional - so run `test-connection` if resource group properties are not set, and only run `test-resource-groups-db` if they are set.

Reply:
Apologies for the oversight.

Adding `"helm.sh/hook-weight": "99"` unconditionally to the `test-connection` test inadvertently affected the `graceful-shutdown` test, which also relies on `helm.sh/hook-weight`. This disrupted the intended order, causing `test-connection` to run after `graceful-shutdown` completes, resulting in a failure.

I’ve adjusted it so that `"helm.sh/hook-weight": "99"` is only applied to `test-connection` during the `resource-groups-properties` test.

Reply:
There are no dependencies between the test cases.

Reply:
You're using this test to setup resource management for the existing connection test. This creates a dependency. We should avoid this.

Reply:
I've removed the `helm.sh/hook-weight` that was conditionally added to `charts/trino/templates/tests/test-connection.yaml` for the `resource-groups-properties` test.

In `charts/trino/templates/tests/test-resource-groups-db.yaml`, I've set `helm.sh/hook-weight` to -1, so it runs before others with the default value of 0.

The existing `charts/trino/templates/tests/test-graceful-shutdown.yaml` also uses `helm.sh/hook-weight` to control execution order, but I believe these configurations do not imply dependencies between tests.

Does that approach seem appropriate to you?

Reply:
No, adjusting hook weights doesn't change that one test depends on the other one. We might have missed this with the graceful shutdown test, but we should not be adding new tests like this.

Either put the setup in a condition block inside `test-connection`, or disable all tests when `.Values.resourceGroups.type` is set to `properties` and add an assert here.

Reply:
Certainly, I understand that test cases should ideally be independent, and that new tests should avoid introducing dependencies. However, in this case, we need to run `test-resource-groups-db` to set up the necessary data before executing queries in `test-connection`.

I believe the approach of "disable all tests when `.Values.resourceGroups.type` is set to `properties` and add an assert here" would require extensive modifications.

Therefore, as a response to the alternative suggestion to "put the setup in a condition block inside `test-connection`," I previously implemented the following in `charts/trino/templates/tests/test-connection.yaml`, which conditionally added a hook weight when `.Values.resourceGroups.type` is set to `"properties`":

https://github.com/trinodb/charts/blob/0a1953e97e76ae59a4af4f7c66a4c5c4f456ec70/charts/trino/templates/tests/test-connection.yaml#L9-L16

Would re-adding this approach be acceptable in this context?

Reply:
No, it's still a dependency between tests. Why don't you want to execute these setup queries in `test-connection`? I can try doing this myself, we've been going back and forth for too long.

Reply:
I just pushed an update to your branch, where I moved the init code to an `initContainer` in the `test-connection` test. I also replaced mysql with postgresql, as we might use it for other tests in near future. I squashed all commits, preparing to merge this. Hope you don't mind.